templates/cluster-template-internal.yaml

```diff
@@ -117,80 +114,93 @@ spec:
 apiVersion: cluster.x-k8s.io/v1beta1
 kind: MachineDeployment
 metadata:
-  name: ${CLUSTER_NAME}-md
+  name: "${CLUSTER_NAME}-md-0"
 spec:
-  clusterName: ${CLUSTER_NAME}
+  clusterName: "${CLUSTER_NAME}"
   replicas: ${WORKER_MACHINE_COUNT}
-  selector: {}
+  selector:
+    matchLabels: {}
   template:
     spec:
-      clusterName: ${CLUSTER_NAME}
-      version: ${KUBERNETES_VERSION}
+      clusterName: "${CLUSTER_NAME}"
+      version: "${KUBERNETES_VERSION:=1.24.0}"
       bootstrap:
         configRef:
           apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
           kind: KubeadmConfigTemplate
-          name: ${CLUSTER_NAME}-md
+          name: "${CLUSTER_NAME}-md-0"
       infrastructureRef:
         apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
         kind: VirtinkMachineTemplate
-        name: ${CLUSTER_NAME}-md
+        name: "${CLUSTER_NAME}-md-0"
 ---
 apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
 kind: KubeadmConfigTemplate
 metadata:
-  name: ${CLUSTER_NAME}-md
+  name: "${CLUSTER_NAME}-md-0"
 spec:
   template:
     spec:
       joinConfiguration:
         nodeRegistration:
           kubeletExtraArgs:
-            provider-id: virtink://{{ ds.meta_data.instance_id }}
+            provider-id: "virtink://{{ ds.meta_data.instance_id }}"
           ignorePreflightErrors:
             - SystemVerification
 ---
 apiVersion: infrastructure.cluster.x-k8s.io/v1beta1
 kind: VirtinkMachineTemplate
 metadata:
-  name: ${CLUSTER_NAME}-md
+  name: "${CLUSTER_NAME}-md-0"
 spec:
   template:
     spec:
       vmSpec:
+        affinity:
+          podAntiAffinity:
+            preferredDuringSchedulingIgnoredDuringExecution:
+              - weight: 100
+                podAffinityTerm:
+                  topologyKey: kubernetes.io/hostname
+                  labelSelector:
+                    matchExpressions:
+                      - key: cluster.x-k8s.io/deployment-name
+                        operator: In
+                        values:
+                          - "${CLUSTER_NAME}-md-0"
```

Comment:
It's necessary to deploy worker in different node?

Reply:
I think so, otherwise a host node restart may cause all workers to fail